### PR TITLE
Drop requirement on boost by using C++17

### PR DESCRIPTION
**Pull request by msarehn**
Drops the requirement on boost by upping the C++ standard to C++17.

Simplifies deployment and packaging.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,7 +11,7 @@ add_executable(peldd
   )
 set_property(TARGET peldd PROPERTY INCLUDE_DIRECTORIES
     ${CMAKE_CURRENT_SOURCE_DIR}/pe-parse/pe-parser-library/include)
-target_link_libraries(peldd ${Boost_LIBRARIES})
+target_link_libraries(peldd)
```
Comment (gsauthof) [inline CMakeLists.txt]: This line can now completely be removed, right?

File: peldd.cc [hunk 2]
```diff
@@ -4,26 +4,35 @@
 #include <parser-library/parse.h>
 #include <parser-library/nt-headers.h>
 
+#include <cctype>
```
Comment (gsauthof) [inline peldd.cc]: For which code is this header needed?

**Review by gsauthof — CHANGES_REQUESTED** [review]
Thank you for your pull-request!

Removing the boost dependency definitely makes sense now since `std::filesystem` was standardized. And just pulling it in for `to_lower_copy()` is overkill.

Thus, I'm in favor of merging your pull request.

File: CMakeLists.txt (hunk 1 above)
Comment (msarehn) [reply CMakeLists.txt]: Seems so. I thought that it wouldn't link at all if it wasn't present, but that seems not to be the case. Removed it.

File: peldd.cc (hunk 2 above)
Comment (msarehn) [reply peldd.cc]: Doesn't seem to be needed at all. Removed it.

**Review by gsauthof — APPROVED** [review]
Excellent change!

**Pull request merged**